docs/design/fabric-v2+/fpc-lifecycle-v2-new.puml

```diff
@@ -0,0 +1,139 @@
+/'
+    Copyright 2020 Intel Corporation
+    Copyright IBM Corp. All Rights Reserved.
+
+    SPDX-License-Identifier: Apache-2.0
+'/
+
+@startuml
+
+!pragma teoz true
+
+hide footbox
+title Foot Box removed
+
+title Fabric Private Chaincode - Lifecycle v2
+
+box "Org1"
+actor	Admin order 10
+entity "FPC_Peer1_Wrapper" as FPC_Peer1_Wrapper order 15
+participant Peer1 order 20
+entity "FPC_Peer2_Wrapper" as FPC_Peer2_Wrapper order 30
+participant Peer2 order 40
+end box
+
+box "Channel Abstraction"
+participant Channel order 70
+note over Channel
+    This box abstracts all other orgs and components.
+    Actions are supposed to be performed by channel participants.
+end note
+end box
+
+entity Orderer order 80
+
+note over Admin
+    Admin uses the fpc-peer wrappers, which abstract the Fabric Peer CLI.
+end note
+
+group chaincode package
+Admin -> FPC_Peer1_Wrapper : lifecycle chaincode package
+note right
+    **Chaincode package**
+    as tar.gz with
+    metadata.json = <path:(enclaves_path), type:(cc_lang), label:(cc_label), sgx_mode: (sim/hw)>
+    code.tar.gz : compressed file with enclave binary and mrenclave
+end note
+return
+end /' group chaincode package '/
+
+group install
+Admin -> FPC_Peer1_Wrapper ++: lifecycle chaincode install
+FPC_Peer1_Wrapper -> Peer1 ++: lifecycle chaincode install
+return packageId
+note over FPC_Peer1_Wrapper
+    Bookeeping operations to take into account that:
+    - FPC's initial version will run a chaincode on a designated peer
+    - the install lifecyle op is peer-specific
+    - the approveformyorg and commit ops are org-specific
+    So the approve stores data for an FPC chaincode installed locally,
+    and the commit checks for data of a locally-approved FPC chaincode.    
+end note
+FPC_Peer1_Wrapper -> FPC_Peer1_Wrapper : retrieve cc_language from package
+note over FPC_Peer1_Wrapper
+    Queryinstalled is redudant, yet consistent with implementation.    
+end note
+FPC_Peer1_Wrapper -> Peer1 ++: lifecycle chaincode queryinstalled
+return packageId
+FPC_Peer1_Wrapper -> FPC_Peer1_Wrapper : if cc_language==fpc then store packageId
```

Comment:
where is the packageId actually stored?

Reply:
on the peer locally? just thinking about implementation .... if someone uses the peer cli to perform `peer chaincode install` remotely ... can this actually work?

Reply:
why do we store packageId? I cant see in this diagram how we use it later.

Reply:
@mbrandenburger : `packageid` has only peer-local semantics. The mapping package-id and the related packages is stored inside the peer. However, for our wrapper to work we also have to keep some mapping which we currently do via files on the filesystem (see also above comments of mine related bookkeeping.